### PR TITLE
Repository

### PR DESCRIPTION
**Pull request by segheim**

### CODE REVIEW

File: tetrahedron/src/main/java/com/epam/jwd/App.java [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
+//
+//
+//        ShapeTetrahedronFactory shapeTetrahedronFactory = new TetrahedronFactory();
+//        TetrahedronFactory tetrahedronFactory = new TetrahedronFactory();
+//        List<Tetrahedron> tetrahedrons = tetrahedronFactory.createTetrahedrons();
+//        System.out.println(tetrahedrons);
+//        for (Tetrahedron tetrahedron : tetrahedrons) {
+//            TetrahedronRepositoryServiceFactory.getTetrahedronRepositoryService().create(tetrahedron);
+//        }
+//        System.out.println(TetrahedronRepositoryServiceFactory.getTetrahedronRepositoryService().getSortByYCoordinateFirstDot());
```
Comment (DmitryGhoncharko) [inline tetrahedron/src/main/java/com/epam/jwd/App.java]: Брат, удали закоменченный код

File: tetrahedron/src/test/java/com/epam/jwd/repository/TetrahedronRepositoryServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package com.epam.jwd.repository;
+
+public class TetrahedronRepositoryServiceTest {
+}
```
Comment (DmitryGhoncharko) [inline tetrahedron/src/test/java/com/epam/jwd/repository/TetrahedronRepositoryServiceTest.java]: brother, your test class without tests

File: tetrahedron/src/main/java/com/epam/jwd/action/SurfaceAreaTetrahedron.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package com.epam.jwd.action;
 
+import com.epam.jwd.entity.ShapeTetrahedron;
```
Comment (DmitryGhoncharko) [inline tetrahedron/src/main/java/com/epam/jwd/action/SurfaceAreaTetrahedron.java:3]: brother, i be make this class as singletone

File: tetrahedron/src/main/java/com/epam/jwd/additional/TetrahedronFactory.java [hunk 4]
```diff
@@ -39,9 +49,12 @@ public List createDotsFromFile() throws IOException {
...
-            LOG.info("Tetrahedron's dots: {}",dotsTetrahedron);
-        }
+            LOG.info("Tetrahedron's dots: {}", dotsTetrahedron);
 
-        return dotsTetrahedron;
+            if (!dotsTetrahedron.isEmpty()) {
+                tetrahedrons.add(new Tetrahedron(dotsTetrahedron));
+            }
+        }
+        return tetrahedrons;
```
Comment (DmitryGhoncharko) [inline tetrahedron/src/main/java/com/epam/jwd/additional/TetrahedronFactory.java]: brother, maybe decompose your code - get data from your reader , validate this data and after put in arguments your factory class(method), but i dont think that my opinion 100% true

File: tetrahedron/src/main/java/com/epam/jwd/register/RegisterTetrahedron.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        register.put(tetrahedron.getTetrahedronId(), calculateParameters(tetrahedron));
+        LOG.info("Add tetrahedron to register {}", tetrahedron);
+
+    }
+
+    private List<Double> calculateParameters(ShapeTetrahedron tetrahedron) {
+
+        List<Double> parameters = new ArrayList<>();
+        SurfaceAreaTetrahedron surfaceAreaTetrahedron = new SurfaceAreaTetrahedron();
+        Double surfaceArea = surfaceAreaTetrahedron.calculateSurfaceAreaTetrahedron((Tetrahedron)tetrahedron);
```
Comment (DmitryGhoncharko) [inline tetrahedron/src/main/java/com/epam/jwd/register/RegisterTetrahedron.java]: broher, i think make acion classes singletone and in registrator recalculation method only call action methods

File: tetrahedron/src/main/java/com/epam/jwd/validations/Validation.java [hunk 6]
```diff
@@ -10,20 +9,23 @@
...
         if (dotsData.length != 4) {
             throw new TetrahedronException();
         }
         return true;
     }
 
     private static boolean isCoordinatesDot(String[] dotsData) throws TetrahedronException {
+
         for (int i = 0; i < dotsData.length; i++) {
             String[] dotCoordinates = dotsData[i].replaceAll("[,;()]", " ").trim().split(" ");
```
Comment (DmitryGhoncharko) [inline tetrahedron/src/main/java/com/epam/jwd/validations/Validation.java:30]: brother, create a regularEx for this

**Review by DmitryGhoncharko — COMMENTED** [review]
this is very nice code, sry but i haven't need qualification for review )

File: tetrahedron/data/info.txt [hunk 7]
```diff
@@ -3,4 +3,9 @@
```
Comment (norka1989) [inline tetrahedron/data/info.txt]: all the code should be placed immediately in the branch, not in the folder  tetrahedron

Comment (norka1989) [reply tetrahedron/data/info.txt]: if i understand correct our teacher)

File: tetrahedron/src/main/java/com/epam/jwd/App.java (hunk 1 above)
Comment (chirlyx) [reply tetrahedron/src/main/java/com/epam/jwd/App.java]: There should be no main in our hometask at all

**Review by chirlyx — COMMENTED** [review]
The project shouldn't be placed in a folder, it's redundant. Also, couldn't find realization of singletone pattern, for example for tetrahedron entity class or its service classes. In general quite nice, I guess.

**Pull request closed without merging**